pkg/datamover/datamover.go

```diff
@@ -0,0 +1,51 @@
+package datamover
+
+import (
+	"github.com/spf13/cobra"
+)
+
+type DataMover interface {
+	// Pull is used to restore the data from object storage
+	// using the preferred data-mover
+	Pull(sourcePath, destinationPath string) error
+	// Push is used to backup the data to object storage
+	// using the preferred data-mover
+	Push(sourcePath, destinationPath string) error
+	// Delete is used to delete the data from object storage
+	// using the preferred data-mover
+	Delete(destinationPath string) error
+}
+
+// NewDataMover creates an instance of DataMover Interface and returns
+// the preferred DataMover as per the arguments passed in kando command
+func NewDataMover(c *cobra.Command) DataMover {
+	datamover := checkDataMover(c)
+	switch datamover {
+	case profileFlagName:
+		outputName := outputNameFlag(c)
+		profileRef, err := unmarshalProfileFlag(c)
+		if err != nil {
+			return nil
+		}
+		snapJSON := kopiaSnapshotFlag(c)
```

Comment:
```suggestion
		snapJSON := kopiaSnapshotFlag(c)
```
can we have this out of switch case, since it is being used for both the datamover.

Reply:
Will Do that !